### PR TITLE
Implement missing part for GPIO interrupts

### PR DESCRIPTION
**Pull request by HakanPayat**
This adds missing parts for GPIO interrupts that a required to setup for pins to be usable.

### CODE REVIEW

File: src/gpio.rs [hunk 1]
```diff
@@ -9,6 +9,32 @@ pub trait GlbExt {
...
+#[derive(Copy, Clone)]
+pub enum Event {
+    /// Trigger on the falling edge
+    NegativePulse = 0,
+    /// Trigger on the rising edge
+    PositivePulse = 1,
+    /// Trigger while low level
+    NegativeLevel = 2,
+    /// Trigger while high level
+    HighLevel = 3
```
Comment (9names) [inline src/gpio.rs:22]: Can you add a trailing comma here, otherwise CI will reject this.
```rust
    HighLevel = 3,

```

**Review by 9names — COMMENTED** [review]
Looks good to me.
Can you merge the latest main into this branch so we can verify CI is happy?

File: src/gpio.rs [hunk 2]
```diff
@@ -9,6 +9,32 @@ pub trait GlbExt {
     fn split(self) -> Parts;
 }
 
+#[allow(non_camel_case_types)]
```
Comment (bjoernQ) [inline src/gpio.rs:12]: is that really needed - looks camel case to me?

Comment (HakanPayat) [reply src/gpio.rs:12]: No, it is a left over before I renamed it. I will remove that.

**Comment by bjoernQ** [comment]
>  Can you merge the latest main into this branch so we can verify CI is happy?

otherwise looks fine to me - also the example looks good

**Comment by HakanPayat** [comment]
I forgot to merge the changes into my fork. It should be ok now for the CI unless I'm still missing something.

**Review by 9names — APPROVED** [review]
Yep, CI is happy.
Thanks for the contribution, going to merge this now.

**Pull request merged**